test/meson.build

```diff
@@ -0,0 +1,106 @@
+
+valgrind_native_args = [
+  '--error-exitcode=1',
+  '--exit-on-first-error=yes',
+  '--leak-check=full',
+  '--quiet',
+  '--show-leak-kinds=all',
+  '--track-origins=yes',
+  '--suppressions=' + ( meson.current_source_dir() / 'valgrind.supp' ),
+]
+
+unit_tests_sources = [
+  'unit-tests.c',
+  'mocks.c',
+  '../lib/dma.c',
+  '../lib/irq.c',
+  '../lib/libvfio-user.c',
+  '../lib/migration.c',
+  '../lib/pci.c',
+  '../lib/pci_caps.c',
+  '../lib/tran.c',
+  '../lib/tran_pipe.c',
+  '../lib/tran_sock.c',
+]
+
+unit_tests_deps = [
+  json_c_dep,
+  cmocka_dep,
+  dl_dep,
+]
+unit_tests_cflags = [
+  '-DUNIT_TEST',
+  '-DWITH_TRAN_PIPE',
+]
+
+unit_tests = executable(
+  'unit_tests',
+  unit_tests_sources,
+  c_args: unit_tests_cflags + common_cflags,
+  dependencies: unit_tests_deps,
+  include_directories: public_include_dir + lib_include_dir,
+  install: false,
+)
+
+test(
+  'unit_tests',
+  unit_tests,
+  suite: 'native',
+)
+
+if opt_valgrind
+  test(
+    'valgrind-unit_tests',
+    valgrind,
+    is_parallel : false,
```

Comment:
I'm not clear why we need this?

Reply:
valgrind is pretty slow when the tests ran it parallel I found the tests were taking long enough that they hit the test timeout. I guess I coud just increase the test timeout instead actually, as that'll probably still be faster overall.

Reply:
So that's not related to meson right?